mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -640,4 +740,55 @@ private object RandomForestSuite {
     val (indices, values) = map.toSeq.sortBy(_._1).unzip
     Vectors.sparse(size, indices.toArray, values.toArray)
   }
+
+  /** Generate a label. */
+  private def generateLabel(rnd: Random, numClasses: Int): Double = {
+    rnd.nextInt(numClasses)
+  }
+
+  /** Generate a numeric value in the range [numericMin, numericMax]. */
+  private def generateNumericValue(rnd: Random, numericMin: Double, numericMax: Double) : Double = {
+    rnd.nextDouble() * (numericMax- numericMin) + numericMin
+  }
+
+  /** Generate a binary value. */
+  private def generateBinaryValue(rnd: Random) : Double = if (rnd.nextBoolean()) 1 else 0
```

Comment:
These 1-line single-use methods strike me as overhead you don't need; I don't feel strongly about it but inlining it would be at least as readable IMHO. This could also be `rnd.nextInt(2)`

Reply:
I have removed _generateBinaryValue_ and used _nextInt_ in place of "_nextBoolean + if_".

I have also in-lined the label generation as it was a similar case.

However, I would keep _generateNumericValue_ as a separate method, as I am afraid it would harm readability.